### PR TITLE
feat: request user agent

### PR DESCRIPTION
**Pull request by yasinpalizban**
hi dear, i add new functionality to request object called reqeust.userAgent() to oak server. this  new feature used to detect operation system.

### CODE REVIEW

**Comment by kitsonk** [comment]
Is this all your original work, or was `useragent.ts` adapted from something?

At this stage of maturity, I would rather focus on getting a user agent library in the Deno [`std/http`](https://github.com/denoland/deno_std/tree/main/http) and then incorporating that into oak once landed.

**Comment by yasinpalizban** [comment]
Hi  my friend 

Here 's story a weeks age  i  try to  detect windows 11  on  request send to server  which was fail to recognize  vis express-userganet  i find out the  repo i used  "express-useragent"    can  not  identify window 11. i did some investigation  and  i got this clue that  window 11 only detect via header.  this  is   microsoft  information about it. [https://learn.microsoft.com/en-us/microsoft-edge/web-platform/how-to-detect-win11](https://github.com/biggora/express-useragent/pull/url) 

 So  I fork express-agent and add feature detect window 11 via header  to original  project  after that  the idea  a cross my mind about why Oak framework  dose not have  user-agent?!. finally. i used my previous fork   and  new changes  guide  me to  fork oak server.   


To sum up, my curiosity about how  identify window 11 push me to fork express-useragent and oak framework  too.

**Comment by kitsonk** [comment]
Ok, there is not attribution of the original project, and you signed a contributor license agreement that legally stated this was your work. That is something we shouldn't do. While the original project that you copied from was MIT licensed, which means it could be adapted and adopted by oak, the code style and the rest of the changes aren't consistent with the project. So respectfully closing.

While it might seem helpful, just copying other people's work and submitting it effectively as your own is not only legally problematic for projects like this, it is just not really ethical. In most cases with projects it is best to raise an issue of the feature you would like to see and let a discussion happen about approach before attempting to submit code, especially code that isn't fully your own work.

A user agent parser does feel like a useful thing though, and I will take a look at contributing something to `std/http` in the near future and then incorporating it into oak.

Sadly this contribution is not suitable for this project, so closing.
